languages/java/src/main/java/net/minecraftforge/fml/javafmlmod/FMLModContainer.java

```diff
@@ -65,7 +69,44 @@ private void constructMod()
         try
         {
             LOGGER.trace(LOADING, "Loading mod instance {} of type {}", getModId(), modClass.getName());
-            this.modInstance = modClass.getDeclaredConstructor().newInstance();
+            try {
+                // Try noargs constructor first
+                this.modInstance = modClass.getDeclaredConstructor().newInstance();
+            } catch (NoSuchMethodException ignored) {
+                // Otherwise look for constructor that can accept more arguments
+                Map<Class<?>, Object> allowedConstructorArgs = Map.of(
+                        IEventBus.class, eventBus,
+                        ModContainer.class, this);
```

Comment:
Question asked by @PepperCode1: should it be `FMLModContainer` instead?

Reply:
No, `FMLModContainer` is specific to the java language loader. Other languages will provider their own implementation.

Reply:
Hmmm, I'm pretty sure this code is java language loader-specific, so it should be fine here?

Reply:
Sorry, thought this was fmlloader code. Yes in this case we can surely provide the implementation type.

Reply:
Added `FMLModContainer` - mods can choose which one they want.